sources/packages/backend/apps/api/src/testHelpers/testing-modules/provider-helper.ts

```diff
@@ -0,0 +1,30 @@
+import { DiscoveryService } from "@golevelup/nestjs-discovery";
+import { Type } from "@nestjs/common";
+import { TestingModule } from "@nestjs/testing";
+
+/**
+ * Get the provider instance for a NestJS module.
+ * When the same provider class is used for different modules,
+ * there will be one instance for each module. When mocking
+ * provider methods, it is important to get the instance for
+ * the right module.
+ * @param testingModule NestJS testing module.
+ * @param module module class.
+ * @param provider provider class.
+ * @returns the provider instance for the given module.
+ */
+export async function getProviderInstanceForModule<T>(
+  testingModule: TestingModule,
+  module: Type,
+  provider: Type<T>,
+) {
+  const discovery = testingModule.get(DiscoveryService);
```

Comment:
looks good. just a suggestion 
![image](https://user-images.githubusercontent.com/77353155/234681034-5d3093f2-385e-483b-8603-e97d89c68ac7.png)
. maybe we can also add the return type